api/src/test/java/org/apache/iceberg/expressions/TestPredicateBinding.java

```diff
@@ -605,4 +605,161 @@ public void testNotInPredicateBindingConversionToExpression() {
     Expression expr = unbound.bind(struct);
     Assert.assertEquals("Should change NOT_IN to alwaysTrue expression", Expressions.alwaysTrue(), expr);
   }
+
+  @Test
+  public void testStructFields() {
```

Comment:
Sorry to make more work, but do you think we can split these two big test methods ?  I feel there's a lot of independent tests here.